src/front-end/typescript/lib/pages/opportunity/code-with-us/view.tsx

```diff
@@ -98,7 +99,7 @@ const init: component_.page.Init<
         viewerUser && isVendor(viewerUser)
           ? api.proposals.cwu.readExistingProposalForOpportunity(
               opportunityId,
-              (response) => response
+              (response) => valid(response)
```

Comment:
Checking the type signature for `readExistingProposalForOpportunity`:

https://github.com/bcgov/digital_marketplace/blob/aa017d5d19ba37d3ded0f93827fdf75c88e41cb2/src/front-end/typescript/lib/http/api/proposal/code-with-us.ts#L43-L54

It looks like it handles the validation there already, and passes a `CWUProposalSlim | undefined` to the `handleResponse` function.

In this case, I think the fix is to change the `InnerMsg` type in this file on L59 from:

```ts
api.ResponseValidation<CWUProposalSlim, string[]> | null
```

to:

```ts
CWUProposalSlim | null
```

And update the rest of the module accordingly.

Reply:
Ah, the use of `unknown` in the HTTP client libraries is really letting a lot of stuff slip through the cracks :( My apologies